streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -54,19 +54,31 @@
     // of the same topic can be assigned to the same topic.
     private final Map<String, TopicPartition> partitionForTopic;
 
+    public ProcessorStateManager(final TaskId taskId,
+                                 final Collection<TopicPartition> sources,
+                                 final boolean isStandby,
+                                 final StateDirectory stateDirectory,
+                                 final Map<String, String> storeToChangelogTopic,
+                                 final ChangelogReader changelogReader,
+                                 final boolean eosEnabled,
+                                 final LogContext logContext) throws IOException {
+        this(taskId, true, sources, isStandby, stateDirectory, storeToChangelogTopic, changelogReader, eosEnabled, logContext);
+    }
+
     /**
      * @throws ProcessorStateException if the task directory does not exist and could not be created
      * @throws IOException if any severe error happens while creating or locking the state directory
      */
     public ProcessorStateManager(final TaskId taskId,
+                                 final boolean createTaskDir,
                                  final Collection<TopicPartition> sources,
                                  final boolean isStandby,
                                  final StateDirectory stateDirectory,
                                  final Map<String, String> storeToChangelogTopic,
                                  final ChangelogReader changelogReader,
                                  final boolean eosEnabled,
                                  final LogContext logContext) throws IOException {
-        super(stateDirectory.directoryForTask(taskId), eosEnabled);
+        super(stateDirectory.directoryForTask(taskId, createTaskDir), eosEnabled);
```

Comment:
as above

Reply:
As explained above, `StateDirectory` don't have any knowledge about the tasks store type.